elastic/kibana
# [Synthetics] adjust permissions for private locations

## Summary

Resolves https://github.com/elastic/kibana/issues/150725

Adds `Integrations: All` permissions requirement for interacting with Private locations

A user without permissions
<img width="1500" alt="Screen Shot 2023-02-23 at 3 26 15 PM" src="https://user-images.githubusercontent.com/11356435/221023579-adc41aa9-2a57-4d97-89ef-048b757a3442.png">
<img width="354" alt="Screen Shot 2023-02-23 at 3 26 06 PM" src="https://user-images.githubusercontent.com/11356435/221023582-7025f9f7-068d-4d86-af2b-4699f51322d8.png">
<img width="359" alt="Screen Shot 2023-02-23 at 3 26 01 PM" src="https://user-images.githubusercontent.com/11356435/221023585-fb53b3fb-39e6-4340-b2dc-709168e66fc5.png">
<img width="1510" alt="Screen Shot 2023-02-23 at 2 36 02 PM" src="https://user-images.githubusercontent.com/11356435/221023588-582aa339-f14b-405c-86e8-8e22df5b9527.png">
<img width="1521" alt="Screen Shot 2023-02-23 at 1 16 38 PM" src="https://user-images.githubusercontent.com/11356435/221023590-2634a673-90a9-4e52-9725-4a2a50fa400e.png">
<img width="1509" alt="Screen Shot 2023-02-23 at 12 28 08 PM" src="https://user-images.githubusercontent.com/11356435/221023592-7acd7bcc-8dd7-40ba-a563-e84e0a0f1e92.png">

A user with permissions
<img width="1520" alt="Screen Shot 2023-02-23 at 1 22 11 PM" src="https://user-images.githubusercontent.com/11356435/221023589-1cca1bf4-c9b5-4776-887d-73eb07b73862.png">

### Testing

1. Create a user with the following permissions
```
Index: synthetics-*: read
Kibana: Uptime/Synthetics: All. Integrations: All
```
2. Log in as that user
3. Navigate to Private locations in Synthetics settings. Confirm the add agent button is disabled and the need permission notice appears
4. Log in as a super user, create a private location
5. Log out, log in as the regular Uptime user
6. Navigate to Private locations in Synthetics settings. Confirm the create location button is disabled and the need permission notice appears
